### PR TITLE
gdal_python.i - adding overviewLevel option to WarpOptions

### PR DESCRIPTION
**Pull request by imiara**
Hi,
This is my first pull request to gdal :)

I was trying to use -ovr level option for gdalwarp in Python. 
Although it is available in the command line app I couldn't find it in the python API and I think it is missing. This pull request is to add this functionality to the Python API.

I added the support in WarpOptions in gdal.py and it works for me (I've tested only the 'level' option. I added support for AUTO and AUTO-n but didn't test them, so I don't know if I set these correctly).

In this pull request I applied the same change to gdal_python.i as I understand gdal.py is auto generated from this file. 
I didn't test the swig file because I don't know how.

### CODE REVIEW

File: gdal/swig/include/python/gdal_python.i [hunk 1]
```diff
@@ -1356,6 +1358,19 @@ def WarpOptions(options=None, format=None,
             new_options += ['-cvmd', str(metadataConflictValue)]
         if setColorInterpretation:
             new_options += ['-setci']
+        if overviewLevel is not None:
+            if _is_str_or_unicode(overviewLevel):
+                overviewLevelStr = overviewLevel
+            else:
+                overviewLevelStr = ''
+                if overviewLevel is ...:
```
Comment (rouault) [inline gdal/swig/include/python/gdal_python.i]: What is this Python syntax (pardon my ignorance) ? Doesn't seem to work with python 2.7

Comment (imiara) [reply gdal/swig/include/python/gdal_python.i]: ellipsis (...) is a builtin constant:
https://docs.python.org/3/library/constants.html
I used it here to express a default value. 
I'm not sure this convention is widespread so maybe We'll use an empty string or NaN instead (or just remove this if statement and use only "AUTO" as is?

Comment (imiara) [reply gdal/swig/include/python/gdal_python.i]: I've forced push a fix:
Changed the default to be 'AUTO' instead of None.
removed the use of ellipsis

**Comment by rouault** [comment]
@imiara This PR is stalled. Do you intend to address my above comments https://github.com/OSGeo/gdal/pull/1255#pullrequestreview-198622832 ?

**Comment by imiara** [comment]
Sorry, I've missed your comments.
I haven't worked with SWIG before, could you please give me some pointers?

**Comment by rouault** [comment]
> I haven't worked with SWIG before, could you please give me some pointers?

- download https://sourceforge.net/projects/swig/files/swig/swig-3.0.8/swig-3.0.8.tar.gz/download if you haven't swig 3.0.8, configure, make, make install
- put in the PATH the location of the swig binary
- in gdal/swig/python, make generate && make

**Comment by imiara** [comment]
Thanks, I'll look into it next week.

**Comment by rouault** [comment]
Integrated with fixes per ab4def2edd041c62968f18985fbb945e247e32ba. Untested though... Tests welcome in autotest/utilitiest/test_gdalwarp_lib.py

**Comment by imiara** [comment]
Hi @rouault ,
Thanks for integrating the pull request!
Unfortunately I didn't find the time yet to write the tests as you advised, I hope I'll get into it.
Although I read that this pull request was merged into master, I can't find it in the master branch.
Please advise.

**Comment by rouault** [comment]
> I can't find it in the master branch

Did you 'git pull' from master branch ? This is in master per https://github.com/OSGeo/gdal/commit/ab4def2edd041c62968f18985fbb945e247e32ba

**Comment by imiara** [comment]
> > I can't find it in the master branch
> 
> Did you 'git pull' from master branch ? This is in master per [ab4def2](https://github.com/OSGeo/gdal/commit/ab4def2edd041c62968f18985fbb945e247e32ba)

I was sure I did, but now I cloned again and found it, thanks!

**Pull request closed without merging**